core/src/main/java/com/amplifyframework/core/model/ModelSchema.java

```diff
@@ -458,14 +485,15 @@ public Builder pluralName(@Nullable String pluralName) {
         }
 
         /**
-         * Denotes whether this model has owner based authorization which changes the parameters for subscriptions
+         * Denotes authorization rules for this model defined by the @auth directive.
          * e.g. @auth(rules: [{allow: owner}]) on the model in the GraphQL Schema
-         * @param hasOwnerAuthorization true if the model has owner based authorization, false otherwise
+         *
+         * @param authRules list of {@link AuthRule}s for this {@link Model}
          * @return the builder object
          */
         @NonNull
-        public Builder hasOwnerAuthorization(boolean hasOwnerAuthorization) {
-            this.hasOwnerAuthorization = hasOwnerAuthorization;
+        public Builder authRules(List<AuthRule> authRules) {
+            this.authRules = authRules;
```

Comment:
To keep `this.authRules` as non-`null`, maybe change this to:
```
if (authRules != null) {
     this.authRules.addAll(authRules);
}
```
Then on L455 you can mark the declaration as `final`, as well.

Reply:
I've updated to the approach below.  Similar to what you suggested, but matches other similar methods on this `Builder` a bit better.

```
        public Builder authRules(@NonNull List<AuthRule> authRules) {
            Objects.requireNonNull(authRules);
            this.authRules.clear();
            this.authRules.addAll(authRules);
            return this;
        }
```